### PR TITLE
[zoe] Update to v3.5

### PR DESCRIPTION
**Pull request by winsoft666**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/zoe/vcpkg.json [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
   "name": "zoe",
-  "version": "3.3",
+  "version": "3.4",
   "maintainers": "winsoft666 <winsoft666@outlook.com>",
   "description": "C++ File Download Library.",
   "homepage": "https://github.com/winsoft666/zoe",
-  "supports": "!osx & !uwp & !arm & !android",
+  "supports": "!uwp & !arm & !android",
```
Comment (MonicaLiu0311) [inline ports/zoe/vcpkg.json:7]: Upstream begins to support macOS: https://github.com/winsoft666/zoe/releases/tag/v3.4.

**Review by MonicaLiu0311 — DISMISSED** [review]

**Comment by dg0yt** [comment]
Modern macOS means arm64. But arm is marked as unsupported.
Shouldn't this be fixed before merging the update?

File: versions/z-/zoe.json [hunk 2]
```diff
@@ -1,5 +1,15 @@
...
   "versions": [
+    {
+      "git-tree": "fe573416ba1ac210d19a22c1ace9eb3b3a4994d0",
+      "version": "3.5",
+      "port-version": 0
+    },
+    {
+      "git-tree": "af971f67231a4122a1b8524e89bd6c50eaffab5a",
+      "version": "3.4",
+      "port-version": 0
```
Comment (JonLiu1993) [inline versions/z-/zoe.json:11]:
```suggestion
```

**Review by JonLiu1993 — CHANGES_REQUESTED** [review]

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Review by JonLiu1993 — APPROVED** [review]

**Pull request merged**